source/codegen/proto_helpers.py

```diff
@@ -168,10 +107,20 @@ def get_enum_parameters(parameter, parameter_name, parameter_type, is_array, use
   })
   return enum_parameters
 
-def get_parameter_type(parameter, is_array, service_class_prefix):
-  if 'grpc_type' in parameter:
-    parameter_type = parameter['grpc_type']
-  else:
-    parameter_type = get_grpc_type_from_ivi(parameter["type"], is_array, service_class_prefix)
-  return parameter_type
 
+def get_parameter_type(parameter):
+  return parameter['grpc_type']
+
+
+def get_parameters(function):
+  parameter_array = filter_parameters_for_grpc_fields(function['parameters'])
+  input_parameters = [p for p in parameter_array if common_helpers.is_input_parameter(p)]
+  if common_helpers.is_init_method(function):
+    has_session_input = any(param['name'] == 'sessionName' for param in input_parameters)
```

Comment:
An alternative to this which might give a little more flexibility is to have the parameter in functions.py have an extra field for `'is_session_name': True`. Then we could check that here if an input param has `is_session_name` set. If not, then create the `session_name_param` and also add that `'is_session_name': True` bit to it. Then in the codegen for init methods we could rely on finding the first (and only) param that has `is_session_name` set. This also would allow you to have the actual param in functions.py be named 'TaskName' if you want.

Reply:
This had slightly more impact and I added a new test for it. Here's the [update(https://github.com/ni/grpc-device/pull/245/files/a0b6385a04928ddd7523c9f03a2a97c5259ff783..ed1991295e9aa081735f484dfe48af843cd20c10#diff-ae080b50783921b45d76b2b536fb4340b7e5fe6d3adb990956b5c8abe2bc493fL96).

Good idea though. We definitely could end up changing the name.